src/pykoala/data_container.py

```diff
@@ -1018,8 +1018,8 @@ def get_integrated_fibres(self, wavelength_range=None):
             Array containing the integrated variance associated to each fibre.
         """
         if wavelength_range is not None:
-            wave_mask = (self.wavelength >= wavelength_range[0]) & (
-                self.wavelength <= wavelength_range[1]
+            wave_mask = (self.wavelength.value >= wavelength_range[0]) & (
```

Comment:
This forces the input wavelength range to be a float, and not a Quantity. The best way to allow both types is to use the method `check_unit`

```
wave_mask = (self.wavelength >= ancillary.check_unit(wavelength_range[0], self.wavelength.unit) &  (self.wavelength <= ancillary.check_unit(wavelength_range[1], self.wavelength.unit)
```
Alternatively, we can check the values of wave_range before creating the mask

```
wave_range = [ancillary.check_unit(wl_r, self.wavelength.unit) for wl_r in wavelength_range]
```

```

Reply:
The first option has issues with the bitwise operator, but the second works just fine. I'll integrate that one